### PR TITLE
[FX-813] Implement refundPayment for POS

### PR DESCRIPTION
**Pull request by djoksimo**
## What

- makes some slight modifications to the Abstract vault submitters to make them compatible with POS
- implements new refundPayment method
- updates Kdoc comments

## Test Plan

- [ ] Add unit tests for the new refund codepath (in this PR)!
- [ ] E2E tests will be added in the later milestone

## How

Can be merged as-is!

### CODE REVIEW

File: forage-android/src/main/java/com/joinforage/forage/android/pos/PosRefundPaymentRepository.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    posTerminalId = posTerminalId,
+                    refundParams = refundParams
+                )
+            )
+        ) {
+            is ForageApiResponse.Success -> PosRefundVaultResponse.ModelMapper.from(response.data)
+            else -> return response
+        }
+
+        val pollingResponse = pollingService.execute(
```
Comment (dleis612) [inline forage-android/src/main/java/com/joinforage/forage/android/pos/PosRefundPaymentRepository.kt:69]: I thought these were synchronous now?

Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/pos/PosRefundPaymentRepository.kt:69]: @dleis612 can you expand on this?

File: forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt [hunk 2]
```diff
@@ -78,7 +79,7 @@ class ForageTerminalSDK(
     ): ForageApiResponse<String> {
         val logger = createLogger()
         logger.i(
-            "[POS] Tokenizing Payment Method via UI PAN entry",
+            "[POS] Tokenizing Payment Method via UI PAN entry on Terminal $posTerminalId",
             attributes = mapOf(
                 "reusable" to reusable
```
[the remark this answers is not in the document]
Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt]: 8de4ae65c6ee657d68e7fafd00df3603c7f24bfe

File: forage-android/src/main/java/com/joinforage/forage/android/pos/PosRefundPaymentRepository.kt (hunk 1 above)
Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/pos/PosRefundPaymentRepository.kt:69]: Confirmation: https://joinforage.slack.com/archives/C061FR1EW2Y/p1705603286791149?thread_ts=1705602932.012439&cid=C061FR1EW2Y

**Review by dleis612 — APPROVED** [review]

**Pull request merged**